org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingReconciler.java

```diff
@@ -326,6 +307,38 @@ public boolean visit(GuardedPattern node) {
 		}
 	}
 
+	/**
+	 * Add a position with the given range and highlighting iff it does not exist already.
+	 * @param offset The range offset
+	 * @param length The range length
+	 * @param highlighting The highlighting
+	 */
+	private void addPosition(int offset, int length, Highlighting highlighting) {
+		boolean isExisting= false;
+		// TODO: use binary search
```

Comment:
here is still a TODO.

Reply:
This method used to be inside of `PositionCollector` (inner class of `SemanticHighlightingReconciler`) and I moved it out to the outer class `SemanticHighlightingReconciler`. This method has only changed the container but implementation remains the same with all of its comments. The TODO there isn't mine :-) 
I could of course resolve the TODO - it doesn't look difficult but it should go as a separate PR ;-)

Reply:
ok